### ISSUE TITLE
Generate 1D mesh from branches

### ISSUE THREAD

**Issue by grongen** [Deltares/HYDROLIB#31]
Generate 1D network with its 1D mesh based on branches (list of LineStrings)

Required:
-	Generate 1D node discretization based on preferred distance
Wish:
-	Specify branch for discretization, so being able to vary preferred node distance per branch.

The result should fit in the hydrolib.core.io.net file model.

**Comment by grongen** [comment]
Yes, that, and the wish to smooth node discretization, are still missing. You can go ahead and add an option for pipes.

**Comment by xldeltares** [comment]
Hi @arthurvd @grongen We have implemented the option for pipes as follows. Once a user specifies node_distance as infinity, the branches are treated as pipes, i.e. no mid point mesh nodes are added. We do pipe straightening before passing the pipe object to this function. 
![image](https://user-images.githubusercontent.com/71022080/181721726-07e87640-ecad-43fa-a978-118f4a6979e1.png)
cc. @rhutten @hboisgon
